File: csharp/Repository/Validators/Database/UserDatabaseValidator.cs
using Domain.Models;
using Repository.Utils;
using Repository.Validators.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Repository.Validators.Database
{
    public class UserDatabaseValidator : IUserValidator
    {
        private DatabaseUtils utils;

        private static readonly String FIND_USER_BY_USERNAME_SQL_STRING = "select user_id from users where username=@username_param";
        public UserDatabaseValidator(IDictionary<String, String> databaseConnectionProperties)
        {
            utils = new DatabaseUtils(databaseConnectionProperties);
        }

        public void Validate(User model)
        {
            String errors = "";

            if (IsUsernameAlreadyUsed(model.UserName))
            {
                errors = "Username is already used!";
            }
            else
            {
                if (model.UserName.Equals(""))
                    errors += "Username can't be empty!\n";

                if (model.Password.Equals(""))
                    errors += "Password can't be empty!";
            }

            if (!errors.Equals(""))
                throw new IUserValidator.InvalidModelException(errors);
        }

        private Boolean IsUsernameAlreadyUsed(String userName)
        {
            using(var connection = utils.GetConnection())
            using(var command = connection.CreateCommand())
            {
                command.CommandText = FIND_USER_BY_USERNAME_SQL_STRING;

                var usernameParam = command.CreateParameter();
                usernameParam.ParameterName = "@username_param";
                usernameParam.Value = userName;

                command.Parameters.Add(usernameParam);

                using(var reader = command.ExecuteReader())
                {
                    return reader.Read();
                }
            }
        }
    }
}

File: java/repository/src/main/java/validators/database/UserDatabaseValidator.java
package validators.database;

import models.User;
import utils.JdbcUtils;
import validators.interfaces.UserValidator;

import java.sql.SQLException;
import java.util.Properties;

public class UserDatabaseValidator implements UserValidator {
    private JdbcUtils utils;

    private static final String FIND_USER_BY_USERNAME_SQL_STRING = "select user_id from users where username=?";

    public UserDatabaseValidator(Properties databaseConnectionProperties) {
        utils = new JdbcUtils(databaseConnectionProperties);
    }


    @Override
    public void validate(User model) {
        String errors = "";

        if (isUsernameAlreadyUsed(model.getUserName())) {
            errors = "Username is already used!";
        } else {
            if (model.getUserName().equals(""))
                errors = "Username can't be empty!\n";

            if (model.getPassword().equals(""))
                errors += "Password can't be empty!\n";
        }

        if (!errors.equals(""))
            throw new InvalidModelException(errors);
    }

    private Boolean isUsernameAlreadyUsed(String username) {
        try(var connection = utils.getConnection();
            var findPreparedStatement = connection.prepareStatement(FIND_USER_BY_USERNAME_SQL_STRING)) {

            findPreparedStatement.setString(1, username);

            try(var resultSet = findPreparedStatement.executeQuery()) {
                return resultSet.next();
            }
        } catch (SQLException exception) {
            exception.printStackTrace();
            return false;
        }
    }
}
